packages/rich-text/src/helpers/editor.ts

```diff
@@ -240,3 +245,14 @@ export const focus = (editor: RichTextEditor) => {
     }, 0);
   }
 };
+
+export function toggleElement(
+  editor: RichTextEditor,
+  options: ToggleNodeTypeOptions,
+  editorOptions?: Omit<EditorNodesOptions, 'match'>
+) {
+  toggleNodeType(editor, options, editorOptions);
+
+  // We must reset `data` from one element to another
+  Transforms.setNodes(editor, { data: {} });
```

Comment:
I think this is unsafe but I could be wrong. @DanweDE this could break for users who used to rely on having custom `data` (for historical reasons) or ?

Reply:
Correction: Yes, it seems under some circumstances custom data can be allowed. I'll send you some more info.

Reply:
Old editor seems to keep the data property when doing the same action:

![image](https://user-images.githubusercontent.com/954889/156367373-5741aa17-9055-45a4-bf0f-d9aa7bc67dd1.png)